sw/device/lib/base/BUILD

```diff
@@ -272,9 +267,11 @@ cc_library(
     srcs = ["status.c"],
     hdrs = ["status.h"],
     deps = [
-        ":absl_status",
         ":bitfield",
         ":macros",
+        "//sw/device/lib/base/internal:status",
+        "//sw/device/lib/dif:base",
```

Comment:
Can we refactor and move this dependency to `lib/base` so that `lib/base` (and production code that depends on `lib/base`) does not depend on `lib/dif`?

Reply:
The only reason to include `dif:base` is so the polymorphic converter macro `INTO_STATUS(x)` can recognize `dif_result_t` and convert it to `status_t`.

Longer term, I'd like to `s/dif_result_t/status_t/` over the whole codebase.  At that point, including `dif:base` will no longer be necessary.

Reply:
SG. I just realized that was the plan when I saw the definition of `INTO_STATUS`. Created #14089.